.github/workflows/prerelease_macos.yml

```diff
@@ -0,0 +1,119 @@
+name: Prerelease macOS pipeline
+
+on:
+  release:
+    types:
+      - prereleased
+    tags:
+      - '*'
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GO_VERSION: '1.14.4'
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+  GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
+  SNYK_TOKEN: ${{ secrets.CAOS_SNYK_TOKEN }}
+  TAG:  ${{ github.event.release.tag_name }}
+  DOCKER_HUB_ID: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+  DOCKER_HUB_PASSWORD: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+  SCHEMA_BRANCH: ${{ github.event.release.tag_name }}
+  AWS_S3_BUCKET_NAME: "nr-downloads-ohai-testing"
+  AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock-testing"
+  AWS_REGION: "us-east-1"
+  DOCKER_PUBLISH: true
+
+jobs:
+  unit-test-macos:
+    name: macOS unit tests
+    runs-on: macos-11
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{env.GO_VERSION}}
+
+      - name: Running unit tests
+        run: make test-coverage
+
+      - name: Coveralls Parallel
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          path-to-lcov: lcov.info
+          flag-name: run-macos
+          parallel: true
+
+  finish:
+    name: Finish
+    needs: [ unit-test-macos ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Coveralls Finished
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          parallel-finished: true
+
+  packaging-macos:
+    name: Build and upload all artifacts into GH Release assets
+    runs-on: ubuntu-20.04
+    needs: [unit-test-macos]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
+
+      - name: Releasing macos packages
+        run: make ci/prerelease/macos
+
+  publishing-to-s3-macos:
+    name: Publish macos artifacts into s3 staging bucket
+    runs-on: ubuntu-20.04
+    needs: [packaging-macos]
+
+    strategy:
+      max-parallel: 1
+      matrix:
+        assetsType:
+          - "targz"
```

Comment:
I would create one macos file with targz ang later add in the same file all other assets. There is no need to split macos for 2-3 different schemas IMO